statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,626 @@
+
+
+class SurveyQuantile(SurveyStat):
+    """
+    Calculates the quantiles[s] for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the mean on
+    parameter: array-like
+        array of quantiles to calculate for each column
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the quantiles on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    quantile : ndarray
+        The quantile[s] to calculate for each column
+    vcov : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, quantile, cov_method='jack', n_reps=None,
+        center_by='global', bsn=None):
+        self.data = np.asarray(data)
+        if self.data.ndim == 1:
+            self.data = self.data[:, None]
+        super().__init__(design, mse)
+        self.quantile = np.asarray(quantile)
+
+        # give warning if user entered in quantile bigger than one
+        if self.quantile.min() < 0 or self.quantile.max > 1:
+            raise ValueError("quantile[s] should be within [0, 1]")
+        self.n_cw = len(self.design.weights)
+
+        # get quantile[s] for each column
+        self.est = np.empty(self.data.shape[1])
+        # need to call this several times
+        self.stderr = np.empty(self.data.shape[1])
+        if cov_method == "jack":
+            for index in range(self.data.shape[1]):
+                self.est[index], self.stderr[index] = self._jackknife()
+        else:
+            for index in range(self.data.shape[1]):
+                self.est[index], self.stderr[index] = self._bootstrap()
+
+    def _stat(self, weights, col_index):
+        quant_list = []
+        cw = np.cumsum(weights)
+        sorted_data = np.sort(self.data[:, col_index])
+        q = self.quantile.copy() * cw[-1]
+        # find index i such that self.cumsum_weights[i] >= q
+        ind = np.searchsorted(cw, q)
+
+        for i, pos in enumerate(ind):
+            # if searchsorted returns length of list
+            # return last observation
+            if pos in np.array([self.n_cw - 1, self.n_cw]):
+                quant_list.append(sorted_data[-1])
+                continue
+            if cw[pos] == q[i]:
+                quant_list.append((sorted_data[pos] + sorted_data[pos+1]) / 2)
+            else:
+                quant_list.append(sorted_data[pos])
+        return quant_list
+
+
+class SurveyMedian(SurveyQuantile):
+    """
+    Derived class from SurveyQuantile with quantile = [.50]
+    """
+    def __init__(self, SurveyDesign, data, cov_method, mse=False):
+        # sp = super(SurveyMedian, self).__init__(SurveyDesign, data, [50])
+        sp = SurveyQuantile(SurveyDesign, data, [.50], cov_method, mse)
```

Comment:
Something is strange here.  SurveyMedian subclasses SurveyQuantile, but also need to create a SurveyQuantile instance?